awk: add page (#9120)
tldr-pages/tldr: pages/osx/awk.md

# awk

> Find and Replace text within file(s).
> More information: <https://ss64.com/osx/awk.html>.

- Define the input field separator = regular expression fs:

> `awk [ -F fs ]`

Reviewer: We don't use the `[ ]` syntax for optional arguments. Also, could you please write the descriptions in a more descriptive way? Currently, it sounds like a man page, which we try to avoid.

> - Define the input field separator = regular expression fs:
> 
> `awk [ -F fs ]`

Reviewer 2: ```suggestion
- Define an input field separator:

`awk --field-separator {{regular_expression}}`
```

Mandatory fix: remove unsupported syntax: unexpected `[` before `-F`.
Mandatory fix: [long option must be used if it's portable](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax).
Mandatory fix: command syntax is wrong: `--field-separator` doesn't accept one literal value `fs`: placeholder required.

Reviewer 3: [Long options do not exist for `awk` on macOS.](https://keith.github.io/xcode-man-pages/awk.1.html)

Reviewer 2: Suggestion updated.

> - Assign specific values to variables

Reviewer 4: ```suggestion
- Assign specific values to variables:
```

> - Assign values before prog is executed:
> 
> `awk [ -v var=value ]`

Reviewer 2: ```suggestion
- Assign specific values to variables

`awk {{--assign variable1=value1 --assign variable2=value2 ...}}`
```

Mandatory fix: [ellipsis must be used](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#special-cases).

> - Specify a pattern to match within progfile:
> 
> `awk [ 'prog' | -f progfile ]`

Reviewer 2: ```suggestion
- Execute a specific program.

`awk --file {{path/to/file}}`
```

Almost the same issue as above.
Optional fix: `specific` may sound better.
Mandatory fix: [path format is broken](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#paths).

> - Set the maximum size of the input record (MaxRows):
> 
> `awk -mr`

Reviewer 2: ```suggestion
- Set the maximum size of the input record:

`awk -mr {{1..infinity}}`
```

Optional fix: description can be simplified.
Optional fix: `-mr` possibly accepts one non-integer number.

> - Set the maximum number of fields (MaxFields):
> 
> `awk -mf`

Reviewer 2: ```suggestion
- Set the maximum number of fields:

`awk -mf {{1..infinity}}`
```

The same issue as above.
